flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/ApplicationReconciler.java

```diff
@@ -128,18 +130,18 @@ public void reconcile(FlinkDeployment flinkApp, Context context) throws Exceptio
                         desiredJobSpec,
                         status,
                         deployConfig,
-                        upgradeMode == UpgradeMode.LAST_STATE);
+                        // We decide to enforce HA based on how job was previously suspended
+                        lastReconciledSpec.getJob().getUpgradeMode() == UpgradeMode.LAST_STATE);
                 stateAfterReconcile = JobState.RUNNING;
             }
             ReconciliationUtils.updateForSpecReconciliationSuccess(flinkApp, stateAfterReconcile);
             IngressUtils.updateIngressRules(
                     deployMeta, currentDeploySpec, deployConfig, kubernetesClient);
         } else if (ReconciliationUtils.shouldRollBack(
-                flinkService, reconciliationStatus, deployConfig)) {
+                flinkService, reconciliationStatus, observeConfig)) {
```

Comment:
Do we need to use the `deployConfig` here? For example, we change the image along with `kubernetes.operator.deployment.rollback.enabled` from `false` to `true`, the rollback should also work.